tests/unit/ViolationUtilsTest.js

```diff
@@ -0,0 +1,192 @@
+import Onyx from 'react-native-onyx';
+import ViolationsUtils from '@libs/Violations/ViolationsUtils';
+import ONYXKEYS from '@src/ONYXKEYS';
+
+const categoryOutOfPolicyViolation = {
+    name: 'categoryOutOfPolicy',
+    type: 'violation',
+    userMessage: '',
+};
+
+const missingCategoryViolation = {
+    name: 'missingCategory',
+    type: 'violation',
+    userMessage: '',
+};
+
+const tagOutOfPolicyViolation = {
+    name: 'tagOutOfPolicy',
+    type: 'violation',
+    userMessage: '',
+};
+
+const missingTagViolation = {
+    name: 'missingTag',
+    type: 'violation',
+    userMessage: '',
+};
+
+describe('getViolationsOnyxData', () => {
+    let transaction;
+    let transactionViolations;
+    let policyRequiresTags;
+    let policyTags;
+    let policyRequiresCategories;
+    let policyCategories;
+
+    beforeEach(() => {
+        transaction = {transactionID: '123'};
+        transactionViolations = [];
+        policyRequiresTags = false;
+        policyTags = {};
+        policyRequiresCategories = false;
+        policyCategories = {};
+    });
+
+    it('should return an object with correct shape', () => {
+        const result = ViolationsUtils.getViolationsOnyxData(transaction, transactionViolations, policyRequiresTags, policyTags, policyRequiresCategories, policyCategories);
+
+        expect(result).toEqual({
+            onyxMethod: Onyx.METHOD.SET,
+            key: `${ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS}${transaction.transactionID}`,
+            value: transactionViolations,
```

Comment:
This would make the test brittle, if the shape changes.

This also makes it clear that the parameter and return value should be the same, and using the standard jest naming is preferred here.